source/tutorials/cross-compilation.rst

```diff
@@ -0,0 +1,262 @@
+.. _ref-cross-compilation:
+
+Cross-compilation
+=================
+
+Cross-compilation is the act of **compiling code** on the **build platform**
+to the **host platform**, where the compiled **executable runs**. [#]_
+
+It's needed when the host platform has limited resources (such as CPU)
+or when it's not easily accessible for development.
+
+The Nix community has world-class support for cross-compilation,
+after years of hard work from our community.
+
+.. [#] Terminology for cross-compilation platforms differs between build systems.
+       We have chosen to follow 
+       `autoconf terminology <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html>`_.
+
+.. note:: macOS/Darwin is a special case, as not the whole OS is open-source. 
```

Comment:
I think for added clarity, it would be nice to mention it's a special case to use as a build platform.

Reply:
It's it a special case for the host platform?

If macOS is a build platform, you can still cross compile to linux as a target for example.

Reply:
Yeah macos -> true FOSS is fine. I might also move this minor caviat down, as none of the text above talks about what platforms are supported yet or anything like that.

Reply:
Oh I was confused here, wouldn't it also be the case with windows then? As windows is also closed source largely?

Reply:
It's possible to make macOS cross work better; it's just no one did yet. With WIndows it has been done.